TWLight/applications/views.py

```diff
@@ -809,13 +811,33 @@ class ListReadyApplicationsView(CoordinatorsOnly, ListView):
     template_name = 'applications/send.html'
 
     def get_queryset(self):
+        # Base queryset of partners with approved applications
+        base_queryset = Partner.objects.filter(
+                            applications__status=Application.APPROVED
+                            )
+
+        # For each partner in the queryset, count how many applications are
+        # from users with data restriction turned on
+        annotated_queryset = base_queryset.annotate(
+                                not_restricted_count = Count(
+                                    Case(When(~Q(applications__editor__user__groups__name='restricted'), then=1),
```

Comment:
This line in particular is pretty dense. I had to look up Django's (Case expressions)[https://docs.djangoproject.com/en/1.11/ref/models/conditional-expressions/#case], (When objects)[https://docs.djangoproject.com/en/1.11/ref/models/conditional-expressions/#when], and (Q objects)[https://docs.djangoproject.com/en/1.11/topics/db/queries/#complex-lookups-with-q-objects] to understand exactly what it was doing. Considering that this is the only place in the application that those are used, I think it warrants more explicit commentary. You could break this single line up and add those comments. Also, since there's only one When in your Case, I think Case might be extraneous. That's an untested accusation, you might need the None it returns as an else condition.